rfc/2022/2022-07-13 - Platform Wide Documentation vs Component Specific Documentation Locations.md

```diff
@@ -0,0 +1,102 @@
+
+Resources like The Platform Website, Confluence, and design.va.gov are utilized by a broad audience, including VFS Teams, Platform Teams, and other VA Platform Stakeholders. These sites also fulfill domain-specific missions:
+
+* teaching developers about the VA Platform at large
+* recording operational information
+* teaching designers how to build standards-compliant user experiences 
+
+For those reasons, we don't think it is appropriate to move their contents inside the Console UI, but we believe there will be content and links that frequently connect these resources.
+
+Using the Console catalog to solve the problems outlined above will necessarily also include the following:
+
+1. **A clear definition of what counts as “component-specific technical documentation.”** This definition will help teams know what documentation should be stored in the Console catalog versus on the Platform Website or in Confluence.
+    1. Documentation that should live in the Console catalog is:
+        1. specific to a software component, API/service, or piece of infrastructure
+        1. technical information needed for development
+        1. intended for use by only a subset of VA teams or roles
+    1. Examples include:
+        1. Documentation that is internal to a VFS team
+        1. Highly technical documentation about an infrastructure resource
+        1. Documentation for VA APIs and downstream external services
+1. **A clear relationship between the Platform Console and the Platform Website.** We propose that the Console and Website contain numerous references to each other. For example, if a component in the Console catalog has documentation that is relevant to all VFS users, it should have a link to Platform Website documentation on the topic. If the Platform Website has a page that mentions technical documentation that not all VFS teams need, it should link to the relevant documentation in the Console catalog. 
+1. **Storing documentation close to the software project/component.** Documentation for a component should be stored in the same repo as the component (where possible) or in a central documentation location and then ingested into the Console catalog, so it is easy to find.
+1. **A self-service process for adding documentation.** Teams who own software components, services, and resources will register these entities in the Console catalog. They will include references to their technical documentation and spell out both upstream and downstream dependencies. They may also create catalog entries for dependencies (and their documentation) that are external to the VA.
+    1. Teams would create a ```/docs``` folder at the root of their project to contain markdown files and subfolders for organizing their documentation.
```

Comment:
How is the project root defined in a shared repository such as `vets-api`?

Reply:
There are a few options and they start being described here: [How to understand techdocs-ref annotation values](https://backstage.io/docs/features/techdocs/how-to-guides#how-to-understand-techdocs-ref-annotation-values)

1. You can follow the standard directions "When you see dir:., you can translate it to mean:" which tell you how to host the documentation in a ```/docs``` folder alongside a ```/mkdocs.yml``` file and and a ```/docs/index.md``` starting page.
2. They also give you the option to specify a subfolder for everything, "dir:./sub-folder" so a VFS team with an app in the vets-website repo, or a team with an API collection can just move the entire bundle of documentation down into their subfolder in the project, but describe the root of their documentation as living in that path... relative to the root of the entire repository.
3. The third option is, "you can instead specify a URL location reference, the exact value of which will vary based on the source code hosting provider." This has a different start to the annotation, rather than ```dir:``` it would use ```url:``` and look something like this: ```url:https://github.com/backstage/backstage/tree/master/plugins/techdocs-backend/examples/documented-component```. So this could be used to refer to component documentation at an explicit URL in the repository, just with a different protocol/method for retrieving the documentation. It can also be used to potentially point to the GitHub Wiki repository... and that might be helpful for giving people a richer editing interface and another way to approach the documentation.

Reply:
Added to RFC